This patch fixes the non saving nature of plugins prefs.  Found in the
preferences dialog under Protocols->msn/jabber.

The original patch can be found here:
http://www.reaperworld.com/code/gaim/gaim-plugin_prefs_final.diff

and the corresponding links in cvs are

plugin.h:
http://cvs.sourceforge.net/viewcvs.py/gaim/gaim/src/plugin.h?r1=1.24&r2=1.25

gtkprefs.c:
http://cvs.sourceforge.net/viewcvs.py/gaim/gaim/src/gtkprefs.c?r1=1.141&r2=1.139

the viewcvs link contains code from a patch that was applied after the
release but before this patch.  It's only 2 lines so it's easy to get
rid of.

Gary

Index: src/gtkprefs.c
===================================================================
RCS file: /cvsroot/gaim/gaim/src/gtkprefs.c,v
retrieving revision 1.140
diff -u -p -r1.140 gtkprefs.c
--- src/gtkprefs.c	26 Apr 2004 04:16:34 -0000	1.140
+++ src/gtkprefs.c	26 Apr 2004 15:25:22 -0000
@@ -363,6 +363,11 @@ delete_prefs(GtkWidget *asdf, void *gdsa
 
 			prefs_info = GAIM_PLUGIN_UI_INFO(plug);
 
+			if(prefs_info->frame != NULL) {
+				gaim_plugin_pref_frame_destroy(prefs_info->frame);
+				prefs_info->frame = NULL;
+			}
+
 			if(prefs_info->iter != NULL) {
 				g_free(prefs_info->iter);
 				prefs_info->iter = NULL;
@@ -1856,7 +1861,8 @@ static void plugin_load (GtkCellRenderer
 				iter = plugin_iter;
 
 			prefs_info = GAIM_PLUGIN_UI_INFO(plug);
-			pref_frame = gaim_gtk_plugin_pref_create_frame(prefs_info->get_plugin_pref_frame(plug));
+			prefs_info->frame = prefs_info->get_plugin_pref_frame(plug);
+			pref_frame = gaim_gtk_plugin_pref_create_frame(prefs_info->frame);
 
 			if(pref_frame != NULL) {
 				prefs_info->iter = g_new0(GtkTreeIter, 1);
@@ -1894,6 +1900,11 @@ static void plugin_load (GtkCellRenderer
 			prefs_info = GAIM_PLUGIN_UI_INFO(plug);
 
 			if(prefs_info != NULL) {
+				if(prefs_info->frame != NULL) {
+					gaim_plugin_pref_frame_destroy(prefs_info->frame);
+					prefs_info->frame = NULL;
+				}
+
 				if(prefs_info->iter != NULL) {
 					gtk_tree_store_remove(GTK_TREE_STORE(prefstree), prefs_info->iter);
 					g_free(prefs_info->iter);
@@ -2544,13 +2555,11 @@ void prefs_notebook_init() {
 
 			if(GAIM_PLUGIN_HAS_PREF_FRAME(plug)) {
 				GtkWidget *gtk_frame;
-				GaimPluginPrefFrame *pp_frame;
 				GaimPluginUiInfo *prefs_info;
 
 				prefs_info = GAIM_PLUGIN_UI_INFO(plug);
-				pp_frame = prefs_info->get_plugin_pref_frame(plug);
-				gtk_frame = gaim_gtk_plugin_pref_create_frame(pp_frame);
-				gaim_plugin_pref_frame_destroy(pp_frame);
+				prefs_info->frame = prefs_info->get_plugin_pref_frame(plug);
+				gtk_frame = gaim_gtk_plugin_pref_create_frame(prefs_info->frame);
 
 				if(GTK_IS_WIDGET(gtk_frame)) {
 					prefs_info->iter = g_new0(GtkTreeIter, 1);
@@ -2558,6 +2567,12 @@ void prefs_notebook_init() {
 										gtk_frame, prefs_info->iter,
 										(plug->info->type == GAIM_PLUGIN_PROTOCOL) ? &proto_iter : &plugin_iter,
 										notebook_page++);
+				} else if(prefs_info->frame) {
+					/* in the event that there is a pref frame and we can
+					 * not make a widget out of it, we free the
+					 * pluginpref frame --Gary
+					 */
+					gaim_plugin_pref_frame_destroy(prefs_info->frame);
 				}
 			}
 		}
Index: src/plugin.h
===================================================================
RCS file: /cvsroot/gaim/gaim/src/plugin.h,v
retrieving revision 1.24
diff -u -p -r1.24 plugin.h
--- src/plugin.h	22 Apr 2004 01:53:11 -0000	1.24
+++ src/plugin.h	26 Apr 2004 15:25:22 -0000
@@ -87,7 +87,7 @@ struct _GaimPluginInfo
 
 	void *ui_info;
 	void *extra_info;
-	void *prefs_info;
+	GaimPluginUiInfo *prefs_info;
 };
 
 /**
@@ -127,6 +127,7 @@ struct _GaimPluginUiInfo {
 	GaimPluginPrefFrame *(*get_plugin_pref_frame)(GaimPlugin *plugin);
 
 	void *iter;                                           /**< Reserved */
+	GaimPluginPrefFrame *frame;                           /**< Reserved */
 };
 
 #define GAIM_PLUGIN_HAS_PREF_FRAME(plugin) \
